### PR TITLE
Display Nutrition

### PR DESCRIPTION
**Pull request by cocovsack**
Sets up backend and frontend to display data fetched from Joey's nutrition servlet.

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -19,15 +19,90 @@ function getRecipeInfo() {
...
+  this.loopNum = 0;
+  this.period = parseInt(period, 10) || 2000;
+  this.txt = '';
+  this.tick();
+  this.isDeleting = false;
+};
+
+/* Print, wait the period, then delete */
+TxtRotate.prototype.tick = function() {
+  var i = this.loopNum % this.toRotate.length;
```
Comment (ALai2) [inline src/main/webapp/script.js:65]: nit: Maybe change the name from i to something like currentLoopPoint.

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -19,15 +19,90 @@ function getRecipeInfo() {
...
+
+  if (this.isDeleting) {
+    this.txt = fullTxt.substring(0, this.txt.length - 1);
+  } else {
+    this.txt = fullTxt.substring(0, this.txt.length + 1);
+  }
+
+  this.el.innerHTML = '<span class="wrap">' + this.txt + '</span>';
+
+  var that = this;
```
Comment (ALai2) [inline src/main/webapp/script.js:76]: Do you need to have a variable "that"? It seems that you can just directly use "this" instead.

File: src/main/webapp/display.html [hunk 3]
```diff
@@ -8,10 +8,57 @@
...
+      <div class="nutrition-info"> </div>
       <div id="display-recipes"> </div>
     </div>
+    <template id="recipe-template">
+      <div class="recipe-card">
+        <p class="recipe-card-title"></p>
+        <div class="recipe-card-table">
+          <div class="recipe-card-block">
+            <img class="recipe-image"></img>
+            <br/><a></a>
```
Comment (joeychin01) [inline src/main/webapp/display.html:41]: What is the blank anchor for?

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -171,139 +246,139 @@ function getLoginStatus() {
...
+
 /** Creates an element that represents a recipe card */
-// TODO: will change to use html template element
 function createRecipeElement(recipe) {
-  const recipeElement = document.createElement('div');
-  recipeElement.className = 'recipe-card';
-
-  const titleElement = document.createElement('p');
+  var temp = document.querySelector("#recipe-template");;
+  var clone = temp.content.cloneNode(true);
```
Comment (joeychin01) [inline src/main/webapp/script.js:266]: maybe change the name to something more descriptive, showing what it is a clone of

File: src/main/webapp/display.html (hunk 3 above)
Comment (cocovsack) [reply src/main/webapp/display.html:41]: I think part of Alisa's template but not sure.

**Pull request merged**